solution_runner/commands/commands_utils.py

```diff
@@ -0,0 +1,89 @@
+import urllib.parse
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
+from typing import Any
+
+import click
+import requests
+import yaml
+
+from . import consts
+
+
+class PathType(Enum):
+    FILE = 1
+    DIRECTORY = 2
+
+
+def get_setting(key: str) -> Any:
+    """
+    :param key: key to retrieve its value
+    :return: value of `key` which is stored in the configuration file
+    """
+    configuration_file = consts.APP_DATA_DIRECTORY / consts.CONFIGURATION_FILE_NAME
+    try:
+        configuration = yaml.safe_load(configuration_file.read_text())
+    except FileNotFoundError:
+        click.secho(
+            "configuration file doesn't exist. Run config command first", fg="red"
+        )
+        raise
+
+    return configuration[key]
+
+
+def check_path_exists(path: Path, command: click.Command, path_type: str = None):
+    """
+    Abort if path doesn't exist.
+    :param path: path to check
+    :param command: command to be passed to`click.Context` if aborting is needed
+    :param path_type: optional "dir" or "file". If not passed, only check for existence. If passed, check for type.
+    """
+    match path_type:
+        case None:
+            if not path.exists():
+                click.secho(f"{path} doesn't exist", fg="red")
+        case "dir":
+            if not path.is_dir():
+                click.secho(
+                    f"Directory at {path} doesn't exist or isn't a directory", fg="red"
+                )
+        case "file":
+            if not path.is_file():
+                click.secho(f"File at {path} doesn't exist or isn't a file", fg="red")
+        case _:
+            return  # If everything is OK, exit the function before aborting.
+    raise click.Abort()
+
+
+def send_aoc_request(method, endpoint: str, payload=None) -> str:
```

Comment:
Replace with a function per HTTP method. This way the GET function does not need to take an optional payload parameter.

Reply:
Why? This is a utility function, `requests.request()` takes an optional payload too.

Reply:
`requests.request()` is a lower level function, the `requests.get()` does not receive data/payload.